_resources/port1.0/group/python-1.0.tcl

```diff
@@ -447,3 +488,33 @@ pre-livecheck {
         livecheck.type  none
     }
 }
+
+# set-up a test phase
+pre-test {
+    if {[option python.test]} {
+        # return for py-<portname> stub ports, which should not have a test-phase enabled
+        if {[string match py-* [option name]] && ${name} eq ${subport}} {
+            return
+        }
+
+        test.run            yes
+        default test.target {}
+
+        if {[option python.test_framework] eq "pytest"} {
+            default test.args   "-o addopts=''"
+        }
+
+        # set PYTHONPATH depending on whether or not a port is "noarch"
+        if {[option supported_archs] eq "noarch"} {
+            default test.env    "PYTHONPATH=${worksrcpath}/build/lib"
+
+            # if using PEP517, unzip the .whl file to ${worksrcpath}/build/lib
+            if {${python.pep517}} {
+                file mkdir ${worksrcpath}/build/lib
+                system -W ${workpath} "unzip -o *.whl -d ${worksrcpath}/build/lib"
+                }
+        } else {
+            test.env    PYTHONPATH=[glob -nocomplain ${worksrcpath}/build/lib*]
```

Comment:
Seems like there should be a way to generate a single value that will work whether or not the port is noarch.

Reply:
correct, `PYTHONPATH=[glob -nocomplain ${worksrcpath}/build/lib*]` will work in both cases - it just needs to be set always in the `pre-test` phase as the directory doesn't exist until the build has finished.